automerge/src/clock.rs

```diff
@@ -1,30 +1,53 @@
 use crate::types::OpId;
 use fxhash::FxBuildHasher;
-use std::cmp;
 use std::collections::HashMap;
 
-#[derive(Debug, Clone, PartialEq)]
-pub(crate) struct Clock(HashMap<usize, u64, FxBuildHasher>);
+#[derive(Default, Debug, Clone, Copy, PartialEq)]
+pub(crate) struct ClockData {
+    /// Maximum operation counter of the actor at the point in time.
+    pub(crate) max_op: u64,
+    /// Sequence number of the change from this actor.
+    pub(crate) seq: u64,
+}
+
+/// Vector clock mapping actor indices to the max op counter of the changes created by that actor.
+#[derive(Default, Debug, Clone, PartialEq)]
+pub(crate) struct Clock(HashMap<usize, ClockData, FxBuildHasher>);
```

Comment:
Does this perform any differently with a `BTreeMap`? `BTreeMap` feels like a more natural fit here where we have integer keys and most of the time a fairly small amount of data.

Reply:
I've not tried but I'm just leaving that as it was for now. May be worth looking at later if it shows up as a bottleneck.